sycl/include/sycl/ext/oneapi/experimental/cuda/barrier.hpp

```diff
@@ -17,7 +17,9 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl::ext::oneapi::experimental::cuda {
 
 class barrier {
+#ifdef __SYCL_DEVICE_ONLY__
   int64_t state;
+#endif
```

Comment:
This means that `barrier` will have different size on the host and the device. Since it is used in local memory it may not be a problem, but I would feel safer if instead of making the class structure different we instead marks it used (using `(void)state;`) in some function, e.g. `initialize`.